contracts/src/v0.8/functions/dev/1_0_0/FunctionsSubscriptions.sol

```diff
@@ -260,10 +255,26 @@ abstract contract FunctionsSubscriptions is IFunctionsSubscriptions, IERC677Rece
 
   // @inheritdoc IFunctionsSubscriptions
   function getSubscription(uint64 subscriptionId) public view override returns (Subscription memory) {
-    _isExistingSubscription(subscriptionId);
     return s_subscriptions[subscriptionId];
   }
 
+  // @inheritdoc IFunctionsSubscriptions
+  function getSubscriptionsInRange(
+    uint64 subscriptionIdStart,
+    uint64 subscriptionIdEnd
+  ) external view override returns (Subscription[] memory subscriptions) {
+    if (subscriptionIdStart > subscriptionIdEnd || subscriptionIdEnd > s_currentSubscriptionId) {
```

Comment:
I think you want to fail if End is >= current.

Example: No subs at all (current = 0), then start = 0, end = 0 should revert and not try to read s_subscriptions[0].

Reply:
Actually, I think this only fails for zero subs and later works, right?

Reply:
current = the last valid subscriptionId though, so if we fail on >= current then we can never fetch the last one

Reply:
instead how about a separate check for if s_currentSubscriptionId === 0 fail

Reply:
You're right - just zero is special.

Reply:
Though I'm not sure that fetching [0] when it doesn't exist is a problem. It's the same as fetching a range where some subs in the middle got closed. It'll just have empty state

Reply:
That's true.

Reply:
Regardless, I added the 0 check